### PR TITLE
SPM-1469: allow build on aarch64

### PR DESCRIPTION
**Pull request by psegedy**
build on aarch64 requires building [librdkafka](https://github.com/edenhill/librdkafka) from source and building go app with `go build -tags dynamic` to link shared library with [confluent-kafka-go](https://github.com/confluentinc/confluent-kafka-go) 

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -22,6 +22,20 @@ ENV GOPATH=/go \
...
+# https://github.com/confluentinc/confluent-kafka-go/issues/576#issuecomment-1009766473
+RUN if [ "$(uname -m)" == "aarch64" ] ; then \
+        DIR=$(pwd) ; \
+        # build librdkafka from source for aarch64
+        git clone --branch v1.6.2 https://github.com/edenhill/librdkafka.git && cd librdkafka ; \
+        dnf install -y gcc-c++ make cyrus-sasl-devel cyrus-sasl-lib cyrus-sasl-gssapi libcurl-devel ; \
+        ./configure --install-deps ; \
+        make ; \
+        make install ; \
+        ln -s /usr/local/lib/librdkafka.so.1 /usr/lib64/librdkafka.so.1 ; \
```
Comment (MichaelMraka) [inline Dockerfile:35]: I see it also builds static library in librdkafka/src/librdkafka.a, can't you just add it to module path so the rest of the files don't have to deal with dynamic linking magic?

**Review by MichaelMraka — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -22,6 +22,20 @@ ENV GOPATH=/go \
     GOPROXY=https://proxy.golang.org \
     PATH=$PATH:/go/bin
 
+# confluent-kafka-go is not built for aarch64
+# https://github.com/confluentinc/confluent-kafka-go/issues/576#issuecomment-1009766473
+RUN if [ "$(uname -m)" == "aarch64" ] ; then \
+        DIR=$(pwd) ; \
```
[the remark this answers is not in the document]
Comment (psegedy) [reply Dockerfile:28]: nice

File: Dockerfile [hunk 3]
```diff
@@ -22,6 +22,20 @@ ENV GOPATH=/go \
...
+        # build librdkafka from source for aarch64
+        git clone --branch v1.6.2 https://github.com/edenhill/librdkafka.git && cd librdkafka ; \
```
[the remark this answers is not in the document]
Comment (psegedy) [reply Dockerfile:30]: sure

File: Dockerfile [hunk 4]
```diff
@@ -22,6 +22,20 @@ ENV GOPATH=/go \
...
+        dnf install -y gcc-c++ make cyrus-sasl-devel cyrus-sasl-lib cyrus-sasl-gssapi libcurl-devel ; \
```
[the remark this answers is not in the document]
Comment (psegedy) [reply Dockerfile:31]: I didn't need it on aarch64

File: Dockerfile (hunk 1 above)
Comment (psegedy) [reply Dockerfile:35]: yeah, I saw that too but some logs suggested like librdkafka.a doesn't contain everything (if thats possible). Dynamic linking is what confluent-kafka-go readme tells us to do, but I can try to use static library

**Review by MichaelMraka — APPROVED** [review]

**Pull request merged**